ui/src/templates/actions/index.ts

```diff
@@ -217,3 +230,39 @@ export const cloneTemplate = (templateID: string) => async (
     dispatch(notify(copy.cloneTemplateFailed(e)))
   }
 }
+
+export const createResourceFromTemplate = (templateID: string) => async (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const template = await client.templates.get(templateID)
+    const {
+      orgs: {org},
+    } = getState()
+
+    const {
+      content: {
+        data: {type},
+      },
+    } = template
+
+    switch (type) {
+      case TemplateType.Dashboard:
+        return dispatch(
+          createDashboardFromTemplate(template as DashboardTemplate, org.id)
```

Comment:
I found this troublesome to get into type guard. We need a tech debt card to make these types more  compatible or centralized.

- redefining different enums in in the UI and client for `TemplateType` makes content type incompatible.
- there seems to be other incompatibilities because just updating the `TemplateType` in `src/types` to be the one from the client didn't help `Template` being incompatible with `ITemplate` from the client.

Reply:
wondering if `TemplateType` needs to be an enum or would the following smell just as sweet with out the thorns?

`type TempalteType = 'task' | 'dashboard' | 'variable'`